src/signer/interface.ts

```diff
@@ -14,4 +15,22 @@ export abstract class SignerInterface extends Provider {
   public abstract override addTransaction(
     transaction: Transaction
   ): Promise<AddTransactionResponse>;
+
+  /**
+   * Sign an JSON object with the starknet private key and return the signature
+   *
+   * @param json - JSON object to be signed
+   * @returns the signature of the JSON object
+   * @throws {Error} if the JSON object is not a valid JSON
+   */
+  public abstract sign(typedData: TypedData): Promise<Signature>;
```

Comment:
We must discriminate between signing for a transaction and signing an off-chain message and that should be reflected in the `Signer` interface. So I would suggest to rename that method to `signMessage`.